apsuite/commissioning_scripts/measure_idffwd.py

```diff
@@ -107,48 +114,49 @@ def measure_at_phase(self, phase):
         traj1 = self._static_get_trajectory()
 
         # measure response matrix
-        curr = self._correctors.orbitcorr_current_sp
+        curr = self.devices['corr'].orbitcorr_current_sp
         mat = _np.zeros((len(traj0), len(curr)))
         for _, corr_idx in enumerate(curr):
+            # register initial current value
             val0 = curr[corr_idx]
 
             # set negative
-            curr[corr_idx] = val0 - MeasAPUFFWD.CORR_DELTA/2
-            self._correctors.orbitcorr_current = curr
-            _time.sleep(MeasAPUFFWD.WAIT_CORR)
+            curr[corr_idx] = val0 - self.params.corr_delta/2
+            self.devices['corr'].orbitcorr_current = curr
+            _time.sleep(self.params.wait_corr)
             trajn = self._static_get_trajectory()
 
             # set positive
-            curr[corr_idx] = val0 + MeasAPUFFWD.CORR_DELTA/2
-            self._correctors.orbitcorr_current = curr
-            _time.sleep(MeasAPUFFWD.WAIT_CORR)
+            curr[corr_idx] = val0 + self.params.corr_delta/2
+            self.devices['corr'].orbitcorr_current = curr
+            _time.sleep(self.params.wait_corr)
             trajp = self._static_get_trajectory()
 
-            # return to init and register matrix column
+            # return current to init and register matrix column
             curr[corr_idx] = val0
-            self._correctors.orbitcorr_current = curr
-            mat[:, corr_idx] = (trajp - trajn) / MeasAPUFFWD.CORR_DELTA
-        _time.sleep(MeasAPUFFWD.WAIT_CORR)
+            self.devices['corr'].orbitcorr_current = curr
+            mat[:, corr_idx] = (trajp - trajn) / self.params.corr_delta
+        _time.sleep(self.params.wait_corr)
 
-        # find corrector values
+        # find correctors values
         dtraj = traj1 - traj0
-        umat, smat, vmat = _np.linalg.svd(mat, full_matrices=False)
+        umat, smat, vhmat = _np.linalg.svd(mat, full_matrices=False)
         inv_s = 1/smat
-        inv_respm = _np.dot(_np.dot(vmat.T, inv_s), umat.T)
+        inv_respm = _np.dot(_np.dot(vhmat.T, inv_s), umat.T)
         currs_delta = - _np.dot(inv_respm, dtraj)
-        return currs_delta, mat, traj0, traj1, umat, smat, vmat
+        return currs_delta, mat, traj0, traj1, umat, smat, vhmat
 
     def measure(self):
         """."""
         for phase, i in enumerate(self.params.phase):
             currs_delta, *_ = self.measure_at_phase(phase)
             self._ffwd[i, :] += currs_delta
 
-    def move_apu(self, phase):
+    def move(self, phase):
         """."""
-        self._apu.phase = phase
-        while abs(self._apu.phase - phase) > MeasAPUFFWD.APU_PHASE_TOL:
-            _time.sleep(0.1)
+        self.devices['apu'].phase = phase
+        while self.devices['apu'].is_moving:
+            _time.sleep(MeasAPUFFWD._PHASE_SLEEP)
```

Comment:
It is missing to call self.devices['apu'].cmd_move() before while statement

Reply:
true. added.